### PR TITLE
Update to ClimaParams v0.10

### PR DESCRIPTION
**Pull request by nefrathenrici**
This PR updates to the new package once it has been released. Depends on Thermodynamics releasing a new patch version first.

### CODE REVIEW

**Comment by trontrytel** [comment]
Wow, I think you are going for the "Most files modified in all CliMA stack per week" award :)

**Comment by nefrathenrici** [comment]
> Wow, I think you are going for the "Most files modified in all CliMA stack per week" award :)

Turns out that CLIMAParameters was widely used, who knew!
Hopefully these changes will only take one week with these CI wait times :)

**Comment by trontrytel** [comment]
I would like to merge this PR before: https://github.com/CliMA/CloudMicrophysics.jl/pull/341

Also, could we make it a breaking release?

File: Project.toml [hunk 1]
```diff
@@ -1,21 +1,21 @@
 name = "CloudMicrophysics"
 uuid = "6a9e3e04-43cd-43ba-94b9-e8782df3c71b"
 authors = ["Climate Modeling Alliance"]
-version = "0.16.0"
+version = "0.17.0"
```
Comment (trontrytel) [inline Project.toml:4]:
```suggestion
version = "0.18.0"
```

**Review by trontrytel — APPROVED** [review]
I just made a `v0.17` release. So lets increase to `v0.18`. This makes CloudMicrophysics.jl an adult package ;)

**Pull request merged**